client/post-editor/editor-drawer/index.jsx

```diff
@@ -245,8 +246,11 @@ const EditorDrawer = React.createClass( {
 		}
 
 		const { plan } = this.props.site;
+		const { blog_public } = this.props.siteSettings;
+		const isSitePrivate = parseInt( blog_public, 10 ) !== 1;
```

Comment:
IIRC I think I saw that `blog_public` was a string. Maybe that's changed now but that's why the `parseInt` is there.